pkg/kms/service.go

```diff
@@ -77,6 +82,35 @@ func (s *service) CreateKey(keystoreID string, kt kms.KeyType) (string, error) {
 
 // Sign signs a message using the key identified by keyID.
 func (s *service) Sign(keystoreID, keyID string, msg []byte) ([]byte, error) {
+	kh, err := s.getKeyHandle(keystoreID, keyID)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.crypto.Sign(msg, kh)
+}
+
+// Verify verifies a signature for the given message using the key identified by keyID.
+func (s *service) Verify(keystoreID, keyID string, sig, msg []byte) error {
+	kh, err := s.getKeyHandle(keystoreID, keyID)
+	if err != nil {
+		return err
+	}
+
+	pub, err := kh.(*keyset.Handle).Public()
+	if err != nil {
+		return err
+	}
+
+	err = s.crypto.Verify(sig, msg, pub)
+	if err != nil && strings.Contains(err.Error(), "verify msg:") {
```

Comment:
Not sure if you can use it here (since it depends on whether the underlying error is publicly accessible), but there is a built in function in Go called `errors.Is()` that you can use to check for errors that are wrapped in other errors

Reply:
Unfortunately, crypto.Verify API doesn't return a custom error type to denote that signature is not valid, so we can't use `errors.Is()` here. But I would like to provide such ability to consumers of my service and that's why I am constructing `ErrInvalidSignature` error here as part of my API. And it's used on the level above (REST) to distinguish the case of "invalid signature" from other possible errors.

Reply:
Sounds good!